### PR TITLE
improve command help

### PR DESCRIPTION
**Pull request by ix0rai**
TODO
- [x] improve arg parsing

CHANGES:
- adds a big enum for all possible args
- greatly improves help printed on command fail:
image
- adds a "help" command
image
- improves quite a bit of code using the new enum
- edits all command names to use `kebab-case`

### CODE REVIEW

File: enigma-cli/src/main/java/cuchaz/enigma/command/HelpCommand.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	@Override
+	public void run(String... args) throws Exception {
+		StringBuilder help = new StringBuilder();
+		Collection<Command> commands = Main.getCommands().values();
+
+		help.append("Supported commands:").append("\n");
+		for (Command command : commands) {
+			help.append("- ").append(command.name).append("\n");
+			// todo iota please help!
+			//help.append("\t").append(command.description.append("\n");
```
Comment (ix0rai) [inline enigma-cli/src/main/java/cuchaz/enigma/command/HelpCommand.java:21]: @IotaBread I'd like you to write descriptions for the commands so we can do this :)

Comment (ix0rai) [reply enigma-cli/src/main/java/cuchaz/enigma/command/HelpCommand.java]: wrote up descriptions on all the ones I'm familiar with, that leaves four more that I'd like help with

File: enigma-cli/src/main/java/cuchaz/enigma/command/InsertProposedMappingsCommand.java [hunk 2]
```diff
@@ -29,42 +29,44 @@
...
-		String resultFormat = getArg(args, 3, "result-format", true);
-		Path profilePath = getReadablePath(getArg(args, 4, "profile", false));
-
-		run(inJar, source, output, resultFormat, profilePath, null);
+	public String getDescription() {
+		return "Adds all mappings proposed by the provided plugins into the given mappings.";
 	}
 
-	public static void run(Path inJar, Path source, Path output, String resultFormat, @Nullable Path profilePath, @Nullable Iterable<EnigmaPlugin> plugins) throws Exception {
+	public void run(Path inJar, Path source, Path output, String resultFormat, @Nullable Path profilePath, @Nullable Iterable<EnigmaPlugin> plugins) throws Exception {
```
Comment (IotaBread) [inline enigma-cli/src/main/java/cuchaz/enigma/command/InsertProposedMappingsCommand.java:61]: Why the change?

File: enigma-cli/src/main/java/cuchaz/enigma/command/InsertProposedMappingsCommand.java [hunk 3]
```diff
@@ -29,42 +29,44 @@
...
-	public void run(String... args) throws Exception {
-		Path inJar = getReadablePath(getArg(args, 0, "in jar", true));
-		Path source = getReadablePath(getArg(args, 1, "source", true));
-		Path output = getWritablePath(getArg(args, 2, "result", true));
-		String resultFormat = getArg(args, 3, "result-format", true);
-		Path profilePath = getReadablePath(getArg(args, 4, "profile", false));
-
-		run(inJar, source, output, resultFormat, profilePath, null);
+	public String getDescription() {
+		return "Adds all mappings proposed by the provided plugins into the given mappings.";
```
Comment (IotaBread) [inline enigma-cli/src/main/java/cuchaz/enigma/command/InsertProposedMappingsCommand.java:58]: I'd replace "provided plugins" with "plugins in the classpath, or the profile if present", or something along those lines

File: enigma-cli/src/main/java/cuchaz/enigma/command/InvertMappingsCommand.java [hunk 4]
```diff
@@ -15,26 +15,29 @@
...
 	@Override
-	public void run(String... args) throws IOException, MappingParseException {
-		Path source = getReadablePath(getArg(args, 0, "source", true));
-		String resultFormat = getArg(args, 1, "result-format", true);
-		Path result = getWritablePath(getArg(args, 2, "result", true));
-
-		run(source, resultFormat, result);
+	public String getDescription() {
+		// todo
+		return null;
```
Comment (IotaBread) [inline enigma-cli/src/main/java/cuchaz/enigma/command/InvertMappingsCommand.java:40]:
```suggestion
		return "Flips the source names with the destination names, ie. 'class a -> Example' becomes 'class Example -> a'";
```

File: enigma-cli/src/main/java/cuchaz/enigma/command/MapSpecializedMethodsCommand.java [hunk 5]
```diff
@@ -22,34 +22,36 @@
...
-	public void run(String... args) throws IOException, MappingParseException {
-		Path jar = getReadablePath(getArg(args, 0, "jar", true));
-		Path source = getReadablePath(getArg(args, 1, "source", true));
-		String resultFormat = getArg(args, 2, "result-format", true);
-		Path result = getWritablePath(getArg(args, 3, "result", true));
-
-		run(jar, source, resultFormat, result);
+	public String getDescription() {
+		// todo
+		return null;
```
Comment (IotaBread) [inline enigma-cli/src/main/java/cuchaz/enigma/command/MapSpecializedMethodsCommand.java:50]:
```suggestion
		return "Adds names for specialized methods from their corresponding bridge method";
```

File: enigma-cli/src/main/java/cuchaz/enigma/command/Main.java [hunk 6]
```diff
@@ -39,8 +39,10 @@ public static void main(String... args) {
 		} catch (CommandHelpException ex) {
 			Logger.error(ex);
 			logEnigmaInfo();
-			Logger.info("Command {} has encountered an error! Usage:", ex.command.name);
-			printHelp(ex.command);
+			Logger.info("Command {} has encountered an error! Usage:", ex.command.getName());
+			StringBuilder help = new StringBuilder();
+			appendHelp(ex.command, help);
+			Logger.info(help.toString());
```
Comment (IotaBread) [inline enigma-cli/src/main/java/cuchaz/enigma/command/Main.java:45]: This is more user-facing, so I'm not sure if we should be even using a logger here

**Review by IotaBread — COMMENTED** [review]
Tbh I didn't know what some of the commands did either, I just checked their code lol.

File: enigma-cli/src/main/java/cuchaz/enigma/command/Main.java (hunk 6 above)
Comment (ix0rai) [reply enigma-cli/src/main/java/cuchaz/enigma/command/Main.java:45]: I'll address that later!

**Review by IotaBread — APPROVED** [review]
